rust/toolchain.bzl

```diff
@@ -6,7 +6,8 @@ ZIP_PATH = "/usr/bin/zip"
 
 # Utility methods that use the toolchain provider.
 def build_rustc_command(ctx, toolchain, crate_name, crate_type, src, output_dir,
-                         depinfo, output_hash=None, rust_flags=[]):
+                         depinfo, output_hash=None, rust_flags=[],
```

Comment:
I think in an earlier attempt I actually did that, but it ended up being quite complicated. That was a while back though. I can take another crack at it and see if we can centralize the linking logic here instead of spread between the `rust_library` and `rust_binary` rules.